### ISSUE TITLE
Enrich build finished logs

### ISSUE THREAD

**Issue by ghost** [jenkinsci/statistics-gatherer-plugin#37]
We've recently added this plugin in the hopes to put together a Grafana dashboard to show average build times, failure rates and so on. 

We've not had much luck so far, though, due to limitations on how much gets logged. I set about graphing build times for master, but am currently unable to exclude builds that were manually restarted by a developer from some middle-stage. If master fails running our E2E suite, we'll restart it from just that step, which results in an erroneously fast build time for any graph I might put together.

It _would_ be possible to filter these out if we had access to the `buildCause` field, but this is currently only logged at the beginning of the build (and these logs don't have the `duration`, or result that I need).

Basically, I want to be able to run the following:

`jobName: master AND result: SUCCESS AND buildCause: Branch indexing` 

So, I guess this is a feature request - would it be possible to repeat all the fields from the 'build started' logs in the corresponding 'build finished' ones?
